pkg/ibmvpc/parser.go

```diff
@@ -741,6 +795,133 @@ func getNACLconfig(rc *ResourcesContainer,
+// currently assuming only all-to-all connectivity is configured
+// in the analysis, such a config should only focus on connections cross-vpcs
+// should make sure that the internal address ranges per all connected vpcs are disjoint
+func extendToTGWconnectedConfigs(tgws map[string]*TransitGateway, res map[string]*vpcmodel.VPCConfig) error {
+	for _, tgw := range tgws {
+		if len(tgw.vpcs) <= 1 {
+			// skip tgw if it does not connect between at least 2 vpcs
+			continue
+		}
+		newConfig := &vpcmodel.VPCConfig{
+			NameToResource:       map[string]vpcmodel.VPCResourceIntf{},
+			IsMultipleVPCsConfig: true,
+		}
+		var vpcsAddressRanges *common.IPBlock // collect all internal address ranges of involved VPCs
+		nacls := &NaclLayer{VPCResource: vpcmodel.VPCResource{ResourceType: vpcmodel.NaclLayer}}
+		sgs := &SecurityGroupLayer{VPCResource: vpcmodel.VPCResource{ResourceType: vpcmodel.SecurityGroupLayer}}
+		vpcsZones := map[string]*Zone{} // collect all zones from all vpcs
+		for _, vpc := range tgw.vpcs {  // iterate the involved VPCs
+			vpcConfig, ok := res[vpc.ResourceUID]
+			if !ok {
+				return fmt.Errorf("missing vpc config for vpc CRN %s", vpc.ResourceUID)
+			}
+			// merge vpc config to the new "combined" config, used to get conns between vpcs only
+			newConfig.Nodes = append(newConfig.Nodes, vpcConfig.Nodes...)
+			newConfig.NodeSets = append(newConfig.NodeSets, vpcConfig.NodeSets...)
+
+			// FilterResources: merge NACLLayers to a single NACLLayer object, same for sg
+			for _, fr := range vpcConfig.FilterResources {
+				switch layer := fr.(type) {
+				case *NaclLayer:
+					nacls.naclList = append(nacls.naclList, layer.naclList...)
+				case *SecurityGroupLayer:
+					sgs.sgList = append(sgs.sgList, layer.sgList...)
+				default:
+					return fmt.Errorf("unexpected type for filter resource in VPC %s", vpc.ResourceUID)
+				}
+			}
+
+			// omit routing resources -- assuming only internal vpc-to-vpc connectivity is of interest to analyze
+			// TODO: is there a scenario of connectivity from one vpc's vsi to external entity through another vpc's pgw/fip ?
+
+			// simple union for NameToResource map
+			for n, r := range vpcConfig.NameToResource {
+				newConfig.NameToResource[n] = r
+			}
+			if vpcsAddressRanges == nil {
+				vpcsAddressRanges = vpcConfig.VPC.(*VPC).internalAddressRange
+			} else {
+				// currently supporting only disjoint address ranges for the connected VPCs
+				intersection := vpcsAddressRanges.Intersection(vpcConfig.VPC.(*VPC).internalAddressRange)
+				if !intersection.Empty() {
+					fmt.Printf("warning: ignoring TGW %s, as currently not supporting connected VPCs with overlapping address ranges\n",
+						tgw.ResourceName)
+					continue
+				}
+				vpcsAddressRanges = vpcsAddressRanges.Union(vpcConfig.VPC.(*VPC).internalAddressRange)
+			}
+
+			// simple union of vpc zones
+			for name, zone := range vpc.zones {
+				// currently assuming no same zones names in the different vpcs
```

Comment:
How can this affect the results?

Reply:
It cannot, removed (left from older version of the code..)